packages/api/src/graphql/subscription/subscription.queries.ts

```diff
@@ -0,0 +1,277 @@
+  filter: Partial<SubscriptionFilter>
+): Prisma.SubscriptionWhereInput => {
+  if (filter?.paidUntilTo) {
+    const {comparison, date} = filter.paidUntilTo
+    const compare = mapDateFilterToPrisma(comparison)
+
+    return {
+      paidUntil: {
+        [compare]: date
+      }
+    }
+  }
+
+  return {}
+}
+
+const createDeactivationDateFromFilter = (
+  filter: Partial<SubscriptionFilter>
+): Prisma.SubscriptionWhereInput => {
+  if (filter?.deactivationDateFrom) {
+    const {comparison, date} = filter.deactivationDateFrom
+    const compare = mapDateFilterToPrisma(comparison)
+
+    return {
+      deactivation: {
+        is: {
+          date: {
+            [compare]: date
+          }
+        }
+      }
+    }
+  }
+
+  return {}
+}
+
+const createDeactivationDateToFilter = (
+  filter: Partial<SubscriptionFilter>
+): Prisma.SubscriptionWhereInput => {
+  if (filter?.deactivationDateTo) {
+    const {comparison, date} = filter.deactivationDateTo
+    const compare = mapDateFilterToPrisma(comparison)
+
+    return {
+      deactivation: {
+        is: {
+          date: {
+            [compare]: date
+          }
+        }
+      }
+    }
+  }
+
+  return {}
+}
+
+const createDeactivationReasonFilter = (
+  filter: Partial<SubscriptionFilter>
+): Prisma.SubscriptionWhereInput => {
+  if (filter?.deactivationReason) {
+    return {
+      deactivation: {
+        reason: filter.deactivationReason as any
+      }
+    }
+  }
+
+  return {}
+}
+
+const createAutoRenewFilter = (
+  filter: Partial<SubscriptionFilter>
+): Prisma.SubscriptionWhereInput => {
+  if (filter?.autoRenew != null) {
+    return {
+      autoRenew: filter.autoRenew
+    }
+  }
+
+  return {}
+}
+
+const createPaymentPeriodicityFilter = (
+  filter: Partial<SubscriptionFilter>
+): Prisma.SubscriptionWhereInput => {
+  if (filter?.paymentPeriodicity) {
+    return {
+      paymentPeriodicity: filter.paymentPeriodicity
+    }
+  }
+
+  return {}
+}
+
+const createPaymentMethodFilter = (
+  filter: Partial<SubscriptionFilter>
+): Prisma.SubscriptionWhereInput => {
+  if (filter?.paymentMethodID) {
+    return {
+      paymentMethodID: filter.paymentMethodID
+    }
+  }
+
+  return {}
+}
+
+const createMemberPlanFilter = (
+  filter: Partial<SubscriptionFilter>
+): Prisma.SubscriptionWhereInput => {
+  if (filter?.memberPlanID) {
+    return {
+      memberPlanID: filter.memberPlanID
+    }
+  }
+
+  return {}
+}
+
+const createHasAddressFilter = (
+  filter: Partial<SubscriptionFilter>
+): Prisma.SubscriptionWhereInput => {
+  if (filter?.userHasAddress) {
+    return {
+      user: {
+        address: {
+          isSet: true
+        }
+      }
+    }
+  }
+
+  return {}
+}
+
+export const createSubscriptionFilter = (
+  filter: Partial<SubscriptionFilter>
+): Prisma.SubscriptionWhereInput => ({
+  AND: [
+    createStartsAtFromFilter(filter),
+    createStartsAtToFilter(filter),
+    createPaidUntilFromFilter(filter),
+    createPaidUntilToFilter(filter),
```

Comment:
Missing Filter which caused mistakenly credit card charges. We have to bring back the filter like here: https://github.com/wepublish/wepublish/commit/a6b5ced56a008a91658877c5f56a4366f3b444f3

May you please note that in the todos?

Reply:
Anyway we have to compare the commit where you checked-out your branch and the latest stable commit and implement all these changes. Should go in the todos.

Reply:
The filters are still there, just not in the GraphQL query, they have been implemented in the memberContext directly.